packages/js/data/src/setting-options/selectors.ts

```diff
@@ -0,0 +1,213 @@
+/**
+ * External dependencies
+ */
+import createSelector from 'rememo';
+
+/**
+ * Internal dependencies
+ */
+import type {
+	SettingsState,
+	SettingsGroup,
+	Setting,
+	SettingValue,
+} from './types';
+
+/**
+ * Get all groups.
+ *
+ * @param {SettingsState} state - The current state of the settings.
+ * @return {SettingsGroup[]} The list of all groups.
+ */
+export const getGroups = ( state: SettingsState ): SettingsGroup[] =>
+	state.groups;
+
+/**
+ * Get a specific group by ID.
+ *
+ * @param {SettingsState} state   - The current state of the settings.
+ * @param {string}        groupId - The ID of the group to get.
+ * @return {SettingsGroup | undefined} The group if found, otherwise undefined.
+ */
+export const getGroup = (
+	state: SettingsState,
+	groupId: string
+): SettingsGroup | undefined =>
+	state.groups.find( ( group ) => group.id === groupId );
+
+// Ensure we have a consistent empty object to return when there are no settings.
```

Comment:
Is this for checking for referential equality?

Reply:
Yes, selectors need to return a scalar or a consistent reference. https://developer.wordpress.org/news/2024/03/how-to-work-effectively-with-the-useselect-hook